assets/js/arrayProcessingTool.js

```diff
@@ -1,94 +1,187 @@
             }
-            return sum;
+            if (i === this.arr.length - 1) {
+                console.log(this.arr[i] + ' last el');
+                console.log('curS:' + currentSum, 'sum: ' + sum, 'in I');
+                break;
+            }
+            for (let j = i + 1; j < this.arr.length; j++) {
+                console.log(this.arr[i], this.arr[j], currentSum);
+                if ((this.arr[j] < 0) && (Math.abs(this.arr[j]) > currentSum)) {
+                    console.log('- Elem: ' + this.arr[j]);
+                    i = j;
+                    if (currentSum > sum) {
+                        sum = currentSum;
+                    }
+                    break;
+                } else {
+                    console.log('Суммируем');
+                    currentSum += this.arr[j];
+                }
+                console.log('curS:' + currentSum, 'sum: ' + sum, 'in J');
+                if (currentSum > sum) {
+                    sum = currentSum;
+                }
+            }
         }
+        return sum;
     },
-    search:function (){
-        const arrLength=this.arr.length;
-        if(!arrLength){
+    search: function () {
+        let arrTmp = [];
+        arrTmp = this.arr.map(el => {
+            if (!arrTmp.includes(el)) {
+                return el;
+            }
+        })
+        console.log(arrTmp);
+        const arrLength = arrTmp.length;
+        if (!arrLength) {
             console.log('array is empty');
-            return {};
+            return 'array is empty';
         }
-        if (arrLength===1){
-            return {max:this.arr[0],min:this.arr[0],middle:this.arr[0]};
+        let max = arrTmp[0], min = arrTmp[0];
+        let canFindMiddle = arrLength % 2;
+        let mid = canFindMiddle ? arrTmp[0] : '';
+        if (arrLength === 1) {
+            return 'max: ' + arrTmp[0] + ' min: ' + arrTmp[0] + ' middle:' + arrTmp[0];
         }
-        let max=this.arr[0],min=this.arr[0];
-        for (let i=1;i<arrLength;i++){
-            if (this.arr[i]>max){
-                max=this.arr[i];
+        if (arrLength === 2) {
+            if (arrTmp[1] > arrTmp[0]) {
+                return 'max: ' + arrTmp[1] + ' min: ' + arrTmp[0] + ' middle:' + mid;
+            }
+            return 'max: ' + arrTmp[0] + ' min: ' + arrTmp[1] + ' middle:' + mid;
+        }
+        for (let i = 1; i < arrLength; i++) {
+            let lessElementsLength = 0, greatElementsLength = 0;
+            for (let j = 0; j < arrLength && canFindMiddle; j++) {
+                console.log('arrTmp[i]:' + arrTmp[i], 'arrTmp[j]:' + arrTmp[j]);
+                if (j !== i) {
+                    if (arrTmp[j] < arrTmp[i]) {
+                        lessElementsLength++;
+                    } else {
+                        greatElementsLength++;
+                    }
+                }
+            }
+            if (lessElementsLength === greatElementsLength) {
+                mid = arrTmp[i];
```

Comment:
The program returns invalid median for the next input data:

- -1 2 3 -9 (median should be equal to 0.5)

- 11 9 3 5 5 (median should be equal to 5)

- 1 3 5 7 (median should be equal to 4)

Reply:
fixed